src/components/select/filterable-select/filterable-select.component.tsx

```diff
@@ -222,7 +222,10 @@ export const FilterableSelect = React.forwardRef(
       (newFilterText: string, isDeleteEvent: boolean) => {
         setSelectedValue((previousValue) => {
           const trimmed = newFilterText.trimStart();
-          const match = findElementWithMatchingText(trimmed, children);
+          const match = findElementWithMatchingText(
+            trimmed,
+            children
+          ) as React.ReactElement;
```

Comment:
**comment**: using the previous comment's typing should mean `match` is automatically typed as either an `Option` or `undefined`:

```suggestion
          const match = findElementWithMatchingText(
            trimmed,
            children
          );
```